README.md

```diff
@@ -1,11 +1,584 @@
+        10000,              # Y shape
+        10000               # X shape
+    ],
+    "zarr_format": 2
+}
+```
+
+### 2.3. OME-NGFF metadata
+
+**REF:** https://ngff.openmicroscopy.org/0.4/#multiscale-md
+
+`.zattrs`
+```python
+{
+    "multiscales": [
+        {
+            "version": "0.4",
+            "axes": [
+                {"name": "t", "type": "time", "unit": "second"},
+                {"name": "c", "type": "channel"},
+                {"name": "z", "type": "space", "unit": "millimeter"},
+                {"name": "y", "type": "space", "unit": "millimeter"},
+                {"name": "x", "type": "space", "unit": "millimeter"}
+            ],
+            "datasets": [
+                {
+                    "path": "0",
+                    "coordinateTransformations": [{
+                        # the voxel size for the first scale level (0.5 millimeter)
+                        "type": "scale",
+                        "scale": [1.0, 1.0, 0.5, 0.5, 0.5]
+                    }]
+                },
+                {
+                    "path": "1",
+                    "coordinateTransformations": [{
+                        # the voxel size for the second scale level
+                        # (downscaled by a factor of 2 -> 1 millimeter)
+                        "type": "scale",
+                        "scale": [1.0, 1.0, 1.0, 1.0, 1.0]
+                    }]
+                },
+                {
+                    "path": "2",
+                    "coordinateTransformations": [{
+                        # the voxel size for the third scale level
+                        # (downscaled by a factor of 4 -> 2 millimeter)
+                        "type": "scale",
+                        "scale": [1.0, 1.0, 2.0, 2.0, 2.0]
+                    }]
+                }
+            ],
+            "coordinateTransformations": [{
+                # the time unit (0.1 seconds),
+                # which is the same for each scale level
+                "type": "scale",
+                "scale": [0.1, 1.0, 1.0, 1.0, 1.0]
+            }],
+        }
+    ]
+}
+```
+
+**FUTURE CHANGES:** as soon as affine coordinate transforms are integrated
+into the OME-NGFF standard, it will be used to encode both the NIfTI
+qform and sform as valid OME-NGFF metadata. However, only these two
+transforms will be accepted in a valid NIfTI-Zarr file. None of the
+more advanced combinations of affine and nonlinear transforms will be
+accepted. Similarly, only a very specific set of coordinate spaces will
+be accepted.
+
+### 2.4. NIfTI header
+
+**REF**: https://nifti.nimh.nih.gov/pub/dist/src/niftilib/nifti1.h
+
+The nifti header ([v1](https://nifti.nimh.nih.gov/pub/dist/src/niftilib/nifti1.h)
+or [v2](https://nifti.nimh.nih.gov/pub/dist/doc/nifti2.h)) **MUST** be encoded as
+a string using [base64](https://en.wikipedia.org/wiki/Base64) encoding and saved in
+the group-level `.zattrs` under the `["nifti"]["base64"]` key:
```

Comment:
As suggested in https://github.com/bids-standard/bids-specification/issues/1704, I would consider saving the header as a binary array inside the base group instead of a JSON metadata field inside `zattrs`. A NIfTI-1 header could be a scalar array of type `|S348` (`np.array(img.header.binaryblock)`) or an array of bytes 348 long (`np.frombuffer(img.header.binaryblock, dtype='u1')`).

Do you have any thoughts on handling NIfTI extensions?

Reply:
part of the challenge is the interplay between ome-zarr and nifti-zarr. as long as we avoid those we should be ok. perhaps just call the key `niftiheader`. i'm sure @balbasty can come up with some specific form that satisfies both constraints :)

Reply:
I like saving the binary block as an array! We'd indeed have to check that it does not violate OME validation.

For extensions, we could allow the binary block to contain everything up to the image data. It would mean that the header can be 348 or 352 or 352 + extension_size bytes. In that case, probably better to have the array be of type `u1` than `S348`. We should also probably require that no compression be used for this array.

Do we have to store it in `nifti/0/` or could it go directly in `nifti/`?

Would we still keep a JSON version of the header in `.zattrs`?

Do we want to formalize the conversion of nifti extensions to JSON?